x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/RoleReference.java

```diff
@@ -127,9 +120,46 @@ public BytesReference getRoleDescriptorsBytes() {
         public ApiKeyRoleType getRoleType() {
             return roleType;
         }
+    }
+
+    final class CrossClusterApiKeyRoleReference implements RoleReference {
```

Comment:
nit: can add some javadoc to this , especially to disambiguate this from `CrossClusterAccessRoleReference`

nit*2: just a style thing...but I generally have a preference towards more top level classes and fewer nested classes. IMO it results in simpler readability and combined with Javadoc leads to less assumptions about what it does.

Reply:
> can add some javadoc to this , especially to disambiguate this from CrossClusterAccessRoleReference

👍 

> nit*2: ...

For this one, I'd prefer to either leave as is, or split the complete `RoleReference` file into separate class files. Otherwise, it feels neither here nor there. LMK if you feel strongly -- happy to follow up with a separate PR to split all the classes!